test/fakeintake/cmd/client/cmd/filtercontainerimages.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/fakeintake/client"
+)
+
+// NewFilterContainerImagesCommand returns the filter container-images command
+func NewFilterContainerImagesCommand(cl **client.Client) (cmd *cobra.Command) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏
First time I see this `**client.Client` are the two `*` expected? Why do we need this there?
I see that we need to write `(*cl)` when we use it, would it be possible/easier to use a normal pointer so we do not need to dereference it once before using it?

Reply:
The reason why we need a double pointer here comes from the order in which objects are initialized by cobra.

There is a single `client.Client` object in the whole process that it pointed to by this variable: https://github.com/DataDog/datadog-agent/blob/1f3aa6a2eaeb865046ee959f0572098f3759783c/test/fakeintake/cmd/client/cmd/root.go#L18

The issue is that this variable is set here: https://github.com/DataDog/datadog-agent/blob/1f3aa6a2eaeb865046ee959f0572098f3759783c/test/fakeintake/cmd/client/cmd/root.go#L26 when any cobra command is invoked.

But this happens **after** the cobra commands are instantiated here: https://github.com/DataDog/datadog-agent/blob/1f3aa6a2eaeb865046ee959f0572098f3759783c/test/fakeintake/cmd/client/cmd/root.go#L32-L37

That’s why we cannot pass directly the `cl` variable: because we would pass the `nil` pointer that `cl` contains before the execution of `cmd.PersistentPreRunE` function.

The alternative would be to have `var cl *client.Client` declared as a global variable so that it can be accessed by any cobra command without having to pass its reference to the cobra command constructors. But I wanted to avoid global variables.